### PR TITLE
Tiny JS world

### PR DESCRIPTION
**Pull request by Zakkarat**
### Tiny JS world

[Demo](https://zakkarat.github.io/a-tiny-JS-world/index.html) |
[Code base](https://github.com/Zakkarat/a-tiny-JS-world)

The code is submitted in its own sub-directory and in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/Zakkarat/a-tiny-JS-world/index.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+const dog = {
+  name: "Paolo",
+  species: "dog",
+  gender: 'male',
```
Comment (o-msh) [inline submissions/Zakkarat/a-tiny-JS-world/index.js]: Try define constants for repeated properties. But in general:
```
const INHABITANT = {
 species: {
  human: 'human'
 ... and so one
 }
}
```

File: submissions/Zakkarat/a-tiny-JS-world/index.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+}
+const cat = {
+  name: "Sophie",
+  species: "cat",
+  gender: 'female',
+  legs: 4,
+  hands: 0,
+  saying: "Mur-meow?"
+}
+const human = {
```
Comment (o-msh) [inline submissions/Zakkarat/a-tiny-JS-world/index.js]: In accordance with the technical specification you must define dog, cat, woman and man. Also cat-woman is optional. I think you must define `man` here.

File: submissions/Zakkarat/a-tiny-JS-world/index.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+}
+const rat = {
+  name: "Twitch",
+  species: "rat",
+  gender: 'male',
+  legs: 2,
+  hands: 2,
+  saying: "Oh, HELLO!"
+}
+const cat_woman = {
```
Comment (o-msh) [inline submissions/Zakkarat/a-tiny-JS-world/index.js]: Use `camelCase`  for that. IMO `catWoman` will be better.

File: submissions/Zakkarat/a-tiny-JS-world/index.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+// ======== OUTPUT ========
+/* Use print(message) for output.
+   Default tag for message is <pre>. Use print(message,'div') to change containing element tag.
+
+   Message can contain HTML markup. You may also tweak index.html and/or styles.css.
+   However, please, REFRAIN from improving visuals at least until your code is reviewed
+   so code reviewers might focus on a single file that is index.js.
+   */
+
+characters.forEach(function(char) {
```
Comment (o-msh) [inline submissions/Zakkarat/a-tiny-JS-world/index.js]: `[dog, cat, human, rat, cat_woman].forEach` will be better. Also try define ES6 arrow function here.

File: submissions/Zakkarat/a-tiny-JS-world/index.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  if(char.hands > 0){
+  print(`Hey there, my name is <strong>${char.name}</strong> and I am a ${char.species}! My gender is ${char.gender}, in addition I have ${char.legs} legs and ${char.hands} hands. Wait a minute... ${char.saying}`)
```
Comment (o-msh) [inline submissions/Zakkarat/a-tiny-JS-world/index.js]: Please, implement a function that would take an object as an argument and return a formatted string where components are combined employing `Array.join`.

File: submissions/Zakkarat/a-tiny-JS-world/index.js (hunk 3 above)
Comment (Zakkarat) [reply submissions/Zakkarat/a-tiny-JS-world/index.js]: Done!

File: submissions/Zakkarat/a-tiny-JS-world/index.js (hunk 2 above)
Comment (Zakkarat) [reply submissions/Zakkarat/a-tiny-JS-world/index.js]: Done!

File: submissions/Zakkarat/a-tiny-JS-world/index.js (hunk 1 above)
Comment (Zakkarat) [reply submissions/Zakkarat/a-tiny-JS-world/index.js]: Sorry, could not understand this part. Could you clarify a bit?

File: submissions/Zakkarat/a-tiny-JS-world/index.js (hunk 5 above)
Comment (Zakkarat) [reply submissions/Zakkarat/a-tiny-JS-world/index.js]: Sorry, did you mean something like this? 
`arr = ['Hey there, my name is <strong>', char.name,'</strong> and I am a ' ...]`

Comment (o-msh) [reply submissions/Zakkarat/a-tiny-JS-world/index.js]: @Zakkarat  I mean
```javascript
const convertToString = inhabitant => {
    return [
        `Hey there, my name is ${inhabitant.name}`,
        `I am a ${inhabitant.species}`,
        `My gender is ${inhabitant.gender}`
    ].join('; ');
};
[dog, cat, etc...].forEach(inhabitant => print(convertToString(inhabitant)));
```

File: submissions/Zakkarat/a-tiny-JS-world/index.js (hunk 1 above)
Comment (o-msh) [reply submissions/Zakkarat/a-tiny-JS-world/index.js]: @Zakkarat I mean define object with constants, then use it for inhabitants.
```javascript
const INHABITANT = {
    species: {
        dog: 'dog',
        human: 'human',
    },
    genders: {
        male: 'male',
        female: 'female',
    }
};
const dog = {
    name: "Paolo",
    species: INHABITANT.species.dog,
    gender: INHABITANT.genders.male,
    legs: 4,
    hands: 0,
    saying: "Bark",
};
```

File: submissions/Zakkarat/a-tiny-JS-world/index.js (hunk 5 above)
Comment (Zakkarat) [reply submissions/Zakkarat/a-tiny-JS-world/index.js]: Thank you! Done!

File: submissions/Zakkarat/a-tiny-JS-world/index.js (hunk 4 above)
Comment (Zakkarat) [reply submissions/Zakkarat/a-tiny-JS-world/index.js]: Done!

File: submissions/Zakkarat/a-tiny-JS-world/index.js (hunk 1 above)
Comment (Zakkarat) [reply submissions/Zakkarat/a-tiny-JS-world/index.js]: Thank you! Done!

**Pull request merged**